mdnsbridge/mdnsbridgeclient.py

```diff
@@ -86,14 +82,16 @@ def getHrefWithException(self, srv_type, priority=None, api_ver=None, api_proto=
 
         return href
 
-    def _getValidServices(self, srv_type, priority, api_ver=None, api_proto=None):
+    def _getValidServices(self, srv_type, priority, api_ver=None, api_proto=None, api_auth=False):
         current_priority = 99
         valid_services = []
         for service in self.services[srv_type]:
             if api_ver is not None and api_ver not in service["versions"]:
                 continue
             if api_proto is not None and api_proto != service["protocol"]:
                 continue
+            if api_auth != service.get("authorization", False):
```

Comment:
Looks like it'll work, but is there any reason for going against the existing pattern?

Reply:
I think I was thinking backwards compatibility wise (on the off chance we were running an older version of nmoscommon)

Reply:
Might be worth just checking we never call the method with some arguments missing. For api_ver/api_proto it looks like if the parameter is missing the method will return all services, whether they have the correct version/protocol or not. The auth addition doesn't seem to provide this option.